File: news_server/modules/source/infrastructure/getters/sspai.ts
/**
 * 少数派新闻源获取器
 * 从 newsnow/server/sources/sspai.ts 迁移
 */
import type { SourceGetter } from '../../domain/getter/getter'
import { NewsItem } from '../../domain/newsitem/entity'

interface SspaiResponse {
  data: {
    id: number
    title: string
  }[]
}

export const sspaiGetter: SourceGetter = async () => {
  const timestamp = Date.now()
  const limit = 30
  const url = `https://sspai.com/api/v1/article/tag/page/get?limit=${limit}&offset=0&created_at=${timestamp}&tag=%E7%83%AD%E9%97%A8%E6%96%87%E7%AB%A0&released=false`
  
  const response = await fetch(url)
  if (!response.ok) {
    throw new Error(`Failed to fetch Sspai: ${response.statusText}`)
  }

  const res = (await response.json()) as SspaiResponse
  
  return res.data.map((k) =>
    NewsItem.fromEntity({
      id: k.id.toString(),
      title: k.title,
      url: `https://sspai.com/post/${k.id}`,
    })
  )
}

File: modules/source/infrastructure/getters/sspai.go
package getters

import (
	"context"
	"fmt"
	"strconv"
	"time"
)

func (r *Registry) sspai(ctx context.Context) ([]Item, error) {
	url := fmt.Sprintf("https://sspai.com/api/v1/article/tag/page/get?limit=30&offset=0&created_at=%d&tag=热门文章&released=false", time.Now().UnixMilli())
	var res struct {
		Data []struct {
			ID    int    `json:"id"`
			Title string `json:"title"`
		} `json:"data"`
	}
	if err := r.getJSON(ctx, url, nil, &res); err != nil {
		return nil, err
	}
	out := make([]Item, 0, len(res.Data))
	for _, k := range res.Data {
		id := strconv.Itoa(k.ID)
		out = append(out, Item{ID: id, Title: k.Title, URL: "https://sspai.com/post/" + id})
	}
	return out, nil
}
